pkg/cloud/services/networking/router.go

```diff
@@ -190,7 +199,20 @@ func (s *Service) setRouterExternalIPs(openStackCluster *infrav1.OpenStackCluste
 }
 
 func (s *Service) DeleteRouter(openStackCluster *infrav1.OpenStackCluster, clusterName string) error {
-	router, subnet, err := s.getRouter(clusterName)
+	routerName := getRouterName(clusterName)
```

Comment:
how about inject this logic into getRouter function 
but in that function we do the check ? seems it's readable? thoughts?

Reply:
If you inject this logic inside a function, then the check (RouterName != "") will have to be done twice. Inside and outside the getRouter function. Also, the getRouter function will have two arguments routerName and clusterName, which is not very obvious. Since routerName can be empty. And if it's not empty then it will override the clusterName argument. It seems that the current version is more readable. what do you think about it?

Reply:
ok, I am fine with easier logic, thanks for the more info